Add io.github.dfandrich.gpscorrelate

dfandrich opened the pull request.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,63 @@
+                "make check",
+                "mv ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.dfandrich.gpscorrelate.svg",
+                "mv ${FLATPAK_DEST}/share/applications/gpscorrelate.desktop ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop",
+                "sed -i 's/Icon=gpscorrelate-gui/Icon=io.github.dfandrich.gpscorrelate/' ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop"
```
use desktop-file-edit instead

dfandrich replied:
Using desktop-file-install now.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,63 @@
+                "make -j${FLATPAK_BUILDER_N_JOBS} prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale CFLAGS=\"${CFLAGS} -DENABLE_NLS=1\"",
+                "make prefix=${FLATPAK_DEST} localedir=${FLATPAK_DEST}/share/gpscorrelate/locale install install-po install-desktop-file",
+                "make check",
+                "mv ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.dfandrich.gpscorrelate.svg",
+                "mv ${FLATPAK_DEST}/share/applications/gpscorrelate.desktop ${FLATPAK_DEST}/share/applications/io.github.dfandrich.gpscorrelate.desktop",
```
use install instead of mv

dfandrich replied:
I've reworked the install process.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,63 @@
+                    "url": "https://www.exiv2.org/builds/exiv2-0.27.3-Source.tar.gz",
+                    "sha256": "a79f5613812aa21755d578a297874fb59a85101e793edc64ec2c6bd994e3e778"
+                }]
+    },
+	{
```
The whole modules seems to be mis-indented

dfandrich replied:
Fixed. There were a few tabs hidden in there, too.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.metainfo.xml:
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Should be sent upstream if possible

dfandrich replied:
It's in process, but I wanted to build the current version right now.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,63 @@
+  "sdk":"org.freedesktop.Sdk",
+  "command":"gpscorrelate-gui",
+  "finish-args":[
+    "--socket=fallback-x11",
```
You need `"--share=ipc"` as well for x11

dfandrich replied:
Done.

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,63 @@
+  "finish-args":[
+    "--socket=fallback-x11",
+    "--socket=wayland",
+	"--filesystem=home",
```
Please re-indent this; also, what does it need home access for?

dfandrich replied:
I wasn't seeing any files from the chooser without --filesystem=home. But I realize now that there must be something else going on there. The code is using gtk_file_chooser_dialog_new which I assume should be working.

bilelmoussaoui replied:
You need to use a native_file_chooser to use the portals.

bilelmoussaoui commented:
bot, build io.github.dfandrich.gpscorrelate

dfandrich commented:
This script is now also copying the metainfo file into place which I hope will fix the test build.

bilelmoussaoui commented:
bot, build io.github.dfandrich.gpscorrelate

bilelmoussaoui commented on io.github.dfandrich.gpscorrelate.json:
```diff
@@ -0,0 +1,69 @@
+                "cp gpscorrelate.desktop ${FLATPAK_ID}.desktop",
+                "desktop-file-install --vendor= --set-icon=${FLATPAK_ID} --dir=/app/share/applications ${FLATPAK_ID}.desktop",
+                "install -m0644 -Dp gpscorrelate-gui.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg",
+                "ls -l",
```
```suggestion
```

dfandrich replied:
Removed.

dfandrich commented:
I've added an upstream exiv2 patch that should fix the ARM build. I've also added the filesystem capability for the moment until the application can use a file chooser portal.

dfandrich commented:
bot, build io.github.dfandrich.gpscorrelate

dfandrich commented:
Updated the patch for aarch64.

bot, build io.github.dfandrich.gpscorrelate

barthalion commented:
/merge
